### PR TITLE
Portfolio: details popup window

### PR DESCRIPTION
**Pull request by Bennyjoez**
# Implementation of Project Details Popup modal.
#### The changes made include: 💯 
- Create an object with information about the various projects. ✔️ 
- Add cards 💳  dynamically using the information stored on the objects in JavaScript. ✔️ 
- Listen to see project button clicks and show details. ✔️ 
- Create a modal to show project details on click. 🚂  
   > Listen for a click 1️⃣ 
   > Get the id for the card whose button is clicked. 2️⃣ 
   > Search the stored objects for the specific Id.  3️⃣ 
   > Create a details modal using the specific information. 4️⃣ 
- Close the details popup card on clicking the close ❌ button. 
- Updating the object array updates the display. 😄 💯

### CODE REVIEW

File: styles.css [hunk 1]
```diff
@@ -260,6 +266,106 @@ body {
...
+#close:active i {
+  color: red;
+}
+
+.popup-heading {
+  position: relative;
+}
+
+#popup-heading {
+  font-family: 'Roboto', sans-serif;
```
Comment (IngDavidTM) [inline styles.css]: - [ ] Great work on designing the header for your new popup! :+1: However, as per the project specifications, it is necessary to use the same font family that is depicted in the Figma template. Therefore, I suggest you switch the font to `'Crete Round', sans-serif`

File: styles.css [hunk 2]
```diff
@@ -260,6 +266,106 @@ body {
...
+  display: none;
+}
+
+#popup-details-card {
+  height: calc(100% + 2px);
+  padding: 16px;
+  background-color: #fff;
+  overflow-y: scroll;
+  border: 1px solid #dfe1e6;
+}
```
Comment (IngDavidTM) [inline styles.css]: - [ ] Your design for the popup is fantastic! :+1: However, it is important that the popup is responsive, meaning it should adjust properly to both mobile and desktop screen sizes without the need for a scrollbar. 
- [ ] Furthermore, please ensure that the popup is centered in the middle of the page, and the buttons below take all the with from the popup to align with the template. 

**Your design**

Screenshot from 2023-03-09 05-23-17

Screenshot from 2023-03-09 05-38-10


**Figma**

Screenshot from 2023-03-09 05-25-33

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi Team 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 3]
```diff
@@ -100,132 +100,17 @@ <h2 class="projects-title">
       </h2>
       <div class="projects-container">
```
Comment (1GORDON) [inline index.html:101]: Great job team 🤝

- [ ] Just like the previous reviewer pointed out, kindly ensure that your design fits properly on both mobile and desktop screens without using the scroll bar. For this case the vertical scroll.
- Your design
your-design
- Given design
given-design

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Bennyjoez & @osmanwako ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 3 above)
Comment (Bennyjoez) [reply index.html:101]: @1GORDON I appreciate your help however, I feel the scroll bar is needed. It appears to allow the user to get a view of all information regarding a project. Some projects will have longer descriptions. 
Also, you added a comment on a section that is not responsible for the scrollbar could you be more specific in your comment and describe what scrollbar you would like me to take care of?

Comment (Bennyjoez) [reply index.html:101]: The popup card is positioned as expected and includes an auto-scroll for when the content is more than what is shown in the design. There is now no need to use a vertical scroll bar to view the content of the modal. 💯

**Review by Shedrack-Sunday — APPROVED** [review]
# STATUS: APPROVED ♻️♻️
Hi @Bennyjoez 

To highlight 👍🏽
✔️ Navigations work as expected
✔️ No linter errors

Your project is complete! There is nothing else to say other than... it's time to merge it 🚀
Congratulations! 🎉

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**